src/lightning/data/streaming/combined.py

```diff
@@ -43,53 +49,120 @@ def __init__(
             self._weights = [w / sum(weights) for w in weights]
 
         self._iterator: Optional[_CombinedDatasetIterator] = None
+        self._use_streaming_dataloader = False
+        self._num_samples_yielded = None
+        self._current_epoch = 0
+
+    def set_epoch(self, current_epoch: int) -> None:
+        self._current_epoch = current_epoch
+        for dataset in self._datasets:
+            dataset.set_epoch(current_epoch)
+
+    def _check_datasets(self, datasets: List[StreamingDataset]) -> None:
+        if any(not isinstance(d, StreamingDataset) for d in datasets):
+            raise RuntimeError("The provided datasets should be instances of the StreamingDataset.")
+
+    def _set_use_streaming_dataloader(self, use_streaming_dataloader: bool) -> None:
+        # Used to prevent returning num_samples_yielded when using PyTorch DataLoader
+        self._use_streaming_dataloader = use_streaming_dataloader
 
     def __len__(self) -> int:
         assert self._weights
         return int(min([1 / w * len(d) for w, d in zip(self._weights, self._datasets) if w > 0]))
 
     def __iter__(self) -> Iterator[Any]:
         assert self._weights
-        self._iterator = _CombinedDatasetIterator(self._datasets, self._seed, self._weights)
+
+        worker_end = _WorkerEnv.detect()
+
+        num_samples_yielded = None
+
+        if self._num_samples_yielded and worker_end.rank in self._num_samples_yielded:
+            num_samples_yielded = self._num_samples_yielded[worker_end.rank]
```

Comment:
Is the logic here going to work out if I call iter() in main, i.e., I just iterate over the combined dataset directly without a dataloader? Because `worker_env.rank` will be 0 anyway. Have you considered this edge case?